sdk/core/azure-core/src/main/java/com/azure/core/models/CloudEvent.java

```diff
@@ -0,0 +1,470 @@
+        }
+        if (Objects.isNull(data)) {
+            throw LOGGER.logExceptionAsError(new NullPointerException("'data' cannot be null."));
+        }
+        if (Objects.isNull(format)) {
+            throw LOGGER.logExceptionAsError(new NullPointerException("'format' cannot be null."));
+        }
+        this.source = source;
+        this.type = type;
+        if (CloudEventDataFormat.BYTES == format) {
+            this.dataBase64 = Base64.getEncoder().encodeToString(data.toBytes());
+        } else {
+            this.data = data.toString();
+        }
+        this.dataContentType = dataContentType;
+        this.id = UUID.randomUUID().toString();
+        this.specVersion = CloudEvent.SPEC_VERSION;
+    }
+
+    private CloudEvent() {
+        // for deserialization
+    }
+
+    /**
+     * Deserialize a list of {@link CloudEvent CloudEvents} from a JSON string and validate whether any CloudEvents have
+     * null id', 'source', or 'type'. If you want to skip this validation, use {@link #fromString(String, boolean)}.
+     * @param cloudEventsJson the JSON payload containing one or more events.
+     *
+     * @return all of the events in the payload deserialized as {@link CloudEvent CloudEvents}.
+     * @throws NullPointerException if cloudEventsJson is null.
+     * @throws IllegalArgumentException if the input parameter isn't a correct JSON string for a cloud event
+     * or an array of it, or any deserialized CloudEvents have null 'id', 'source', or 'type'.
+     */
+    public static List<CloudEvent> fromString(String cloudEventsJson) {
+        return fromString(cloudEventsJson, false);
+    }
+
+    /**
+     * Deserialize a list of {@link CloudEvent CloudEvents} from a JSON string.
+     * @param cloudEventsJson the JSON payload containing one or more events.
+     * @param skipValidation set to true if you'd like to skip the validation for the deserialized CloudEvents. A valid
+     *                       CloudEvent should have 'id', 'source' and 'type' not null.
+     *
+     * @return all of the events in the payload deserialized as {@link CloudEvent CloudEvents}.
+     * @throws NullPointerException if cloudEventsJson is null.
+     * @throws IllegalArgumentException if the input parameter isn't a JSON string for a cloud event or an array of it,
+     * or skipValidation is false and any CloudEvents have null id', 'source', or 'type'.
+     */
+    public static List<CloudEvent> fromString(String cloudEventsJson, boolean skipValidation) {
+        if (cloudEventsJson == null) {
+            throw LOGGER.logExceptionAsError(new NullPointerException("'cloudEventsJson' cannot be null"));
+        }
+        try {
+            List<CloudEvent> events = Arrays.asList(SERIALIZER.deserialize(
+                new ByteArrayInputStream(cloudEventsJson.getBytes(StandardCharsets.UTF_8)),
+                TypeReference.createInstance(CloudEvent[].class)));
```

Comment:
Should use the generics supported type reference `new TypeReference<List<CloudEvent> { }`, and make it a constant on `CloudEvent`.